### PR TITLE
fix: don't include view stroke by default for non-cartesian plot

### PR DESCRIPTION
**Pull request by kanitw**
fix https://github.com/vega/vega-lite/issues/7621

### CODE REVIEW

File: src/compile/layer.ts [hunk 1]
```diff
@@ -100,6 +101,17 @@ export class LayerModel extends Model {
...
   }
 
+  public assembleGroupStyle(): string | string[] {
+    const uniqueStyles = new Set<string>();
+    for (const child of this.children) {
+      for (const style of array(child.assembleGroupStyle())) {
+        uniqueStyles.add(style);
+      }
+    }
+    const styles = Array.from(uniqueStyles);
```
Comment (domoritz) [inline src/compile/layer.ts:111]: I think you can check the length of the set and return the first item of the set instead. No need to make it an array.

**Review by domoritz — APPROVED** [review]
Please simplify the set code but otherwise looks good.

File: src/compile/layer.ts (hunk 1 above)
Comment (kanitw) [reply src/compile/layer.ts:111]: To get the first item you need to get the iterator and call next, which seems equally expensive as this but result in a more complicated code. So I think this is actually better.

Comment (domoritz) [reply src/compile/layer.ts:111]: I disagree. The array call makes a copy while creating an iterator is highly optimized.

Comment (domoritz) [reply src/compile/layer.ts:111]: I guess since we may return an array, it doesn't make a big difference.

Comment (kanitw) [reply src/compile/layer.ts:111]: Most of the time N=1 or a few.  Maybe iterator is marginally better, but produces way longer code that doesn't justify the benefit.

**Pull request merged**